bookrack-backend/src/test/java/org/example/bookrackbackend/BookControllerTest.java

```diff
@@ -124,4 +116,35 @@ void addBook_shouldReturnSuccess_whenAllFieldsAreProvided() throws Exception {
                         .param("year", "2021"))
                 .andExpect(status().isOk());
     }
+
+    @DirtiesContext
+    @Test
+    void addBook_shouldReturnBadRequest_whenBookWithSameTitleAndAuthorExists() throws Exception {
+        when(cloudinary.uploader()).thenReturn(uploader);
+        when(uploader.upload(any(), any())).thenReturn(Map.of("url", "test"));
```

Comment:
what do you use the returned Map for? I can't find "url" and "test" in your test data. 

Maybe these lines are superflous or you should check the Map entries. (when-commands usually appear in unit tests, rarely in integration tests.)